src/main/java/ttt/Board.java

```diff
@@ -99,52 +59,51 @@ public boolean isFinished() {
         return (isDraw() || isWon());
     }
 
-    public boolean isWon() {
-        for (List<String> selection : winningPositions()) {
-            if (isAWin(selection)) {
-                return true;
-            }
+    public boolean isAWinFor(Marks mark) {
+        for (List<Marks> cells : winningPositions()) {
+            if (Collections.frequency(cells, mark) == 3) return true;
         }
         return false;
     }
 
-    private List<String> getCollection(int first, int middle, int last) {
-        return Arrays.asList(grid[first], grid[middle], grid[last]);
-    }
-
-    private boolean isValidLocation(int location) {
-        return (location <= 8 && location >= 0);
-    }
-
-    private boolean isCellTaken(int location) {
-        return (!grid[location].equals(emptyCell));
+    public boolean isWon() {
+        for (List<Marks> cells : winningPositions()) {
+            if (Collections.frequency(cells, Marks.O) == 3) return true;
+            if (Collections.frequency(cells, Marks.X) == 3) return true;
+        }
+        return false;
```

Comment:
This needs changing, but I can't quite decide how it should be so I'm waiting for 4 x 4 board to force me to find a better way...